web/api/v1/api.go

```diff
@@ -824,12 +824,29 @@ func (api *API) labelValues(r *http.Request) (result apiFuncResult) {
 }
 
 var (
-	// MinTime is the default timestamp used for the begin of optional time ranges.
-	// Exposed to let downstream projects to reference it.
+	// MinTime is the default timestamp used for the start of optional time ranges.
+	// Exposed to let downstream projects reference it.
+	//
+	// The offset used here (62135596801) is inspired by the difference in
+	// seconds between the Unix time zero point (1970-01-01 00:00:00 UTC)
+	// and the Go time zero point (0001-01-01 00:00:00 UTC, following the
+	// Gregorian calendar), which is 62135596800 seconds. The offset is used
+	// to avoid overflow/underflow of an int64 when converting back to
+	// "milliseconds since Unix time epoch", which is what we use as the
+	// timestamp in Prometheus. However, the offset is in fact not needed
+	// for MinTime. It was introduced here in the past by accident, and we
+	// don't want to change the (exported!) MinTime value now. The value is
+	// still low enough for all practical purposes.
 	MinTime = time.Unix(math.MinInt64/1000+62135596801, 0).UTC()
 
 	// MaxTime is the default timestamp used for the end of optional time ranges.
 	// Exposed to let downstream projects to reference it.
+	//
+	// See the note about the offset (62135596801) above, with the
+	// additional note that an offset is indeed needed in this case to avoid
+	// an overflow, but -1 would be sufficient (or alternatively, we could
+	// set the nsec argument to zero). Again, we don't want to change
+	// anything now, as the value is high enough for all practical purposes.
 	MaxTime = time.Unix(math.MaxInt64/1000-62135596801, 999999999).UTC()
```

Comment:
Love it, but some of the details raised more questions actually -- had to do my however to understand fully what overflows you have in mind and why we don't have math.MaxInt64-62135596801 even for max as suggested in https://stackoverflow.com/a/32620397. What's even funnier is that I don't know where +1 to 62135596800 comes from.. (accident likely) 🙈
I would suggest reversing Max first, since "something" like this is indeed there and then adding a playground example for explanation. Created this: https://go.dev/play/p/Mg5A0vLHePr

Also fixing some duplicated mentions.

```suggestion
	// MaxTime is the default timestamp used for the end of optional time ranges.
	// Exposed and used by downstream projects to reference as a sentinel value.
	//
	// The offset used here (62135596801) is inspired by the difference in
	// seconds between the Unix time zero point (1970-01-01 00:00:00 UTC)
	// and the Go time zero point (0001-01-01 00:00:00 UTC, following the
	// Gregorian calendar), which is 62135596800 seconds. 
	//
	// This offset is used to avoid overflow/underflow of an int64 when converting
	// to "milliseconds since Unix time epoch" we use in Prometheus e.g. using
	// `model/timestamp/FromTime`. 
	//
	// There are slightly higher and clearer MaxTime values we could use, 
	// (see https://go.dev/play/p/Mg5A0vLHePr for examples). However, for
	// compatibility, we can't change it and the value is high enough for all practical purposes.
	MaxTime = time.Unix(math.MaxInt64/1000-62135596801, 999999999).UTC()
	
	// MinTime is the default timestamp used for the start of optional time ranges.
	// Exposed and used by downstream projects to reference as a sentinel value.
	//
	//  See MaxTime for the context on 62135596801 offset. In case of MinTime, this
	// offset is not needed and added by accident.  However, for compatibility, we 
	// can't change it and the value is small enough for all practical purposes.
	MinTime = time.Unix(math.MinInt64/1000+62135596801, 0).UTC()
```